plugin/build.gradle

```diff
@@ -42,47 +74,93 @@ android {
-            }
             ndk {
                 //noinspection ChromeOsAbiSupport
                 abiFilters 'arm64-v8a', 'x86_64'
             }
         }
         meta {
             dimension "vendor"
-            externalNativeBuild {
-                cmake {
-                    arguments "-DFLAVOR=meta"
-                }
-            }
         }
         pico {
             dimension "vendor"
-            externalNativeBuild {
-                cmake {
-                    arguments "-DFLAVOR=pico"
-                }
-            }
+        }
+    }
+
+    sourceSets {
+        khronosDebug {
+            jniLibs.srcDirs = [
+                'src/main/libs/debug/arm64-v8a',
+                'src/main/libs/debug/x86_64',
+                '../thirdparty/khronos_openxr_sdk/arm64-v8a',
+                '../thirdparty/khronos_openxr_sdk/x86_64',
+            ]
+        }
+        khronosRelease {
+            jniLibs.srcDirs = [
+                'src/main/libs/release/arm64-v8a',
+                'src/main/libs/release/x86_64',
+                '../thirdparty/khronos_openxr_sdk/arm64-v8a',
+                '../thirdparty/khronos_openxr_sdk/x86_64',
+            ]
+        }
+        lynxDebug {
+            jniLibs.srcDirs = [
+                'src/main/libs/debug/arm64-v8a',
+                '../thirdparty/lynx_openxr_sdk',
+            ]
+        }
+        lynxRelease {
+            jniLibs.srcDirs = [
+                'src/main/libs/release/arm64-v8a',
+                '../thirdparty/lynx_openxr_sdk',
+            ]
+        }
+        magicleapDebug {
+            jniLibs.srcDirs = [
+                'src/main/libs/debug/arm64-v8a',
+                'src/main/libs/debug/x86_64',
+                '../thirdparty/khronos_openxr_sdk/arm64-v8a',
+                '../thirdparty/khronos_openxr_sdk/x86_64',
+            ]
+        }
+        magicleapRelease {
+            jniLibs.srcDirs = [
+                'src/main/libs/release/arm64-v8a',
+                'src/main/libs/release/x86_64',
+                '../thirdparty/khronos_openxr_sdk/arm64-v8a',
+                '../thirdparty/khronos_openxr_sdk/x86_64',
+            ]
+        }
+        metaDebug {
+            jniLibs.srcDirs = [
+                'src/main/libs/debug/arm64-v8a',
+                '../thirdparty/ovr_openxr_mobile_sdk/OpenXR/Libs/Android/Debug',
+            ]
+        }
+        metaRelease {
+            jniLibs.srcDirs = [
+                'src/main/libs/release/arm64-v8a',
+                '../thirdparty/ovr_openxr_mobile_sdk/OpenXR/Libs/Android/Release',
+            ]
+        }
+        picoDebug {
+            jniLibs.srcDirs = [
+                'src/main/libs/debug/arm64-v8a',
+                '../thirdparty/khronos_openxr_sdk/arm64-v8a',
+            ]
+        }
+        picoRelease {
+            jniLibs.srcDirs = [
+                'src/main/libs/release/arm64-v8a',
+                '../thirdparty/khronos_openxr_sdk/arm64-v8a',
+            ]
```

Comment:
Why are we not using the default `jniLibs` directories instead?
Do all the vendors have jni dependencies in the same `src/main/libs` sub-folders?

Reply:
> Why are we not using the default `jniLibs` directories instead?

I don't think we can, because there's no libraries that are shared by every AAR. They do all include arm64 libraries, but since we have to make the distinction between debug and release, they are all different.

> Do all the vendors have jni dependencies in the same `src/main/libs` sub-folders?

They all have some dependencies from there, just not all the same ones.